lib/dependency-manager-adapters/npm.js

```diff
@@ -194,10 +194,13 @@ module.exports = CoreObject.extend({
     let restoreTasks = [
       copy(path.join(this.cwd, this.packageJSONBackupFileName),
         path.join(this.cwd, this.packageJSON)),
-      copy(path.join(this.cwd, this.nodeModulesBackupLocation),
-        path.join(this.cwd, this.nodeModules), { clobber: true }),
     ];
 
+    if (fs.existsSync(path.join(this.cwd, this.nodeModulesBackupLocation))) {
+      restoreTasks.push(copy(path.join(this.cwd, this.nodeModulesBackupLocation),
+        path.join(this.cwd, this.nodeModules), { clobber: true }));
+    }
```

Comment:
Yarn workspaces can hoist packages in such a way that a `node_modules` folder may never be created. This checks the existence of the directory before trying to restore it